### PR TITLE
[UCSC-134] Sidebar Menu

### PR DESCRIPTION
**Pull request by bjcooper**
## What does this do/fix?

Adds styles and JS behaviors for the "Content with left navigation" block pattern.

## QA

Links to relevant issues
- [UCSC-134](https://example.com)](https://moderntribe.atlassian.net/browse/UCSC-134)

Screenshots/video:
Screencast at September 20th 2023 - 1 15 50 pm

### CODE REVIEW

**Comment by bjcooper** [comment]
_bump_ @Vinsanity

File: src/scss/block-patterns/_sidebar-with-navigation.scss [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+			}
+		}
+
+		// Navigation menu
+		> .wp-block-navigation__container {
+			width: 100%;
+			gap: 0;
+
+			// Top-level menu item
+			> .wp-block-navigation-item {
```
Comment (Vinsanity) [inline src/scss/block-patterns/_sidebar-with-navigation.scss:43]: the nesting here is getting quite deep which will create an excessively long string. Can we move this out of here and add classes to blocks to reduce that?

**Review by Vinsanity — CHANGES_REQUESTED** [review]
@bjcooper concept looks great! nice work. Just some small items to address and clean up and one or two items to review for next time. Let me know if you have any questions on those.

File: patterns/sidebar-with-navigation.php [hunk 2]
```diff
@@ -5,17 +5,18 @@
...
-<div class="wp-block-column" style="flex-basis:33.33%"><!-- wp:heading {"level":3,"fontSize":"base"} -->
-<h3 class="wp-block-heading has-base-font-size"><strong>Navigation header</strong></h3>
+<!-- wp:columns {"className":"ucsc__sidebar-with-navigation"} -->
+<div class="wp-block-columns ucsc__sidebar-with-navigation"><!-- wp:column {"width":"33.33%"} -->
+<div class="wp-block-column" style="flex-basis:33.33%"><!-- wp:heading {"level":3,"style":{"typography":{"textTransform":"uppercase"}},"fontSize":"base"} -->
+<h3 class="wp-block-heading has-base-font-size" style="text-transform:uppercase">Navigation header</h3>
 <!-- /wp:heading -->
 
-<!-- wp:navigation { "maxNestingLevel": 1,"overlayMenu":"never","layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} /--></div>
+<!-- wp:navigation {"ref":1857,"textColor":"ucsc-royal-blue","overlayMenu":"never","overlayTextColor":"ucsc-royal-blue","maxNestingLevel":1,"layout":{"type":"flex","orientation":"vertical","justifyContent":"left"},"fontSize":"base"} /--></div>
```
[the remark this answers is not in the document]
Comment (bjcooper) [reply patterns/sidebar-with-navigation.php]: Thanks, Vinny. The `ref` entry has been removed.

File: src/js/block-patterns/sidebar-with-navigation.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+				toggle.setAttribute( 'aria-hidden', ! isExpanded );
+			}
+		} );
+	} );
+};
+
+/**
+ * Bind event listeners.
+ */
+const bindEvents = () => {
```
[the remark this answers is not in the document]
Comment (bjcooper) [reply src/js/block-patterns/sidebar-with-navigation.js:70]: Oh, nice! I've used that in SQ1 projects before but didn't know we had it available in this project.

File: src/js/block-patterns/sidebar-with-navigation.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+			const item = submenu.closest( c.itemSelector );
+			const wrapper = document.createElement( 'div' );
+			wrapper.classList.add( c.submenuWrapperClass );
+			item.append( wrapper );
+			wrapper.append( submenu );
+		} );
+
+		// Remove nested submenus and their toggles.
+		block
+			.querySelectorAll( c.submenuSelector + ' ' + c.submenuSelector )
```
[the remark this answers is not in the document]
Comment (bjcooper) [reply src/js/block-patterns/sidebar-with-navigation.js]: Done.

File: src/js/block-patterns/sidebar-with-navigation.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
+const blocks = document.querySelectorAll( '.ucsc__sidebar-with-navigation' );
+
+// Cache object, populated upon initialization.
+let c = {};
```
[the remark this answers is not in the document]
Comment (bjcooper) [reply src/js/block-patterns/sidebar-with-navigation.js]: I renamed blocks to use the elements convention. The other bits are not cached element references, just selector strings and CSS classes I didn't want to hardcode as magic strings through the script. Those, I just flatly declare now in order to be less confusing.

**Review by Vinsanity — APPROVED** [review]
🟢

**Pull request merged**